From a11295481389cda3dc8130b2863e71f5183d4537 Mon Sep 17 00:00:00 2001
From: Tom Marshall <tdm.code@gmail.com>
Date: Wed, 16 Jan 2019 12:18:04 -0800
Subject: [PATCH] ntfs-3g: Add static libs for recovery

 * Let libfuse-lite and libntfs-3g build statically.
 * Add libntfs_utils for common usage.
 * Add libntfs_mkfs, libntfs_mkfs, libntfs_mount.

Change-Id: I2d49384065eb9b5b5b8ed138e28d6a48f549264e
---
 Android.bp | 62 ++++++++++++++++++++++++++++++++++++++++++++++++++++--
 1 file changed, 60 insertions(+), 2 deletions(-)

diff --git a/Android.bp b/Android.bp
index 5bc5d753..a0bbf008 100644
--- a/Android.bp
+++ b/Android.bp
@@ -28,7 +28,7 @@ cc_defaults {
     tags: ["optional"],
 }
 
-cc_library_shared {
+cc_library {
     name: "libfuse-lite",
 
     srcs: ["libfuse-lite/*.c"],
@@ -37,7 +37,7 @@ cc_library_shared {
     export_include_dirs: ["include/fuse-lite"],
 }
 
-cc_library_shared {
+cc_library {
     name: "libntfs-3g",
 
     srcs: [
@@ -123,3 +123,61 @@ cc_binary {
         "libntfs-3g",
     ],
 }
+
+// Libraries for recovery
+
+cc_library_static {
+    name: "libntfs_utils",
+    defaults: ["ntfs-3g_defaults"],
+    srcs: [
+        "ntfsprogs/utils.c",
+    ],
+    cflags: ["-Dmain=fsck_ntfs_main"],
+    static_libs: [
+        "libfuse-lite",
+        "libntfs-3g",
+    ],
+}
+
+cc_library_static {
+    name: "libntfs_fsck",
+    defaults: ["ntfs-3g_defaults"],
+    srcs: [
+        "ntfsprogs/ntfsfix.c",
+    ],
+    cflags: ["-Dmain=fsck_ntfs_main"],
+    static_libs: [
+        "libfuse-lite",
+        "libntfs-3g",
+    ],
+}
+
+cc_library_static {
+    name: "libntfs_mkfs",
+    defaults: ["ntfs-3g_defaults"],
+    srcs: [
+        "ntfsprogs/attrdef.c",
+        "ntfsprogs/boot.c",
+        "ntfsprogs/mkntfs.c",
+        "ntfsprogs/sd.c",
+    ],
+    cflags: ["-Dmain=mkfs_ntfs_main"],
+    static_libs: [
+        "libfuse-lite",
+        "libntfs-3g",
+    ],
+}
+
+cc_library_static {
+    name: "libntfs_mount",
+    defaults: ["ntfs-3g_defaults"],
+    srcs: [
+        "src/ntfs-3g.c",
+        "src/ntfs-3g_common.c",
+    ],
+    cflags: ["-Dmain=mount_ntfs_main"],
+    static_libs: [
+        "libfuse-lite",
+        "libntfs-3g",
+    ],
+}
-- 
2.17.1

